ingestion/src/metadata/profiler/metrics/system/system.py

```diff
@@ -58,207 +38,81 @@ def recursive_dic():
-                rowsAffected: <int>,
-            } else returns None
-    """
-    logger.debug(f"System metrics not support for {dialect}. Skipping processing.")
-
-
-@get_system_metrics_for_dialect.register(Dialects.BigQuery)
-def _(
-    dialect: str,
-    session: Session,
-    table: DeclarativeMeta,
-    conn_config: BigQueryConnection,
-    *args,
-    **kwargs,
-) -> List[SystemProfile]:
-    """Compute system metrics for bigquery
-
-    Args:
-        dialect (str): bigquery
-        session (Session): session Object
-        table (DeclarativeMeta): orm table
-
-    Returns:
-        List[Dict]:
-    """
-    logger.debug(f"Fetching system metrics for {dialect}")
-
-    project_id = session.get_bind().url.host
-    dataset_id = table.__table_args__["schema"]  # type: ignore
-
-    metric_results: List[Dict] = []
-
-    jobs = get_value_from_cache(
-        SYSTEM_QUERY_RESULT_CACHE, f"{Dialects.BigQuery}.{project_id}.{dataset_id}.jobs"
-    )
-
-    if not jobs:
-        cursor_jobs = session.execute(
-            text(
-                JOBS.format(
-                    usage_location=conn_config.usageLocation,
-                    dataset_id=dataset_id,
-                    project_id=project_id,
-                    insert=DatabaseDMLOperations.INSERT.value,
-                    update=DatabaseDMLOperations.UPDATE.value,
-                    delete=DatabaseDMLOperations.DELETE.value,
-                    merge=DatabaseDMLOperations.MERGE.value,
-                )
-            )
-        )
-        jobs = [
-            BigQueryQueryResult(
-                query_type=row.statement_type,
-                timestamp=row.start_time,
-                table_name=row.destination_table,
-                dml_statistics=row.dml_statistics,
-            )
-            for row in cursor_jobs
-        ]
-        set_cache(
-            SYSTEM_QUERY_RESULT_CACHE,
-            f"{Dialects.BigQuery}.{project_id}.{dataset_id}.jobs",
-            jobs,
+T = TypeVar("T")
+
+
+class CacheProvider(ABC, Generic[T]):
+    def __init__(self):
+        self.cache = LRUCache[T](LRU_CACHE_SIZE)
+
+    def get_or_update_cache(
+        self,
+        cache_path: str,
+        get_queries_fn: Callable[..., List[T]],
+        *args,
+        **kwargs,
+    ):
+        if cache_path in self.cache:
+            return self.cache.get(cache_path)
+        result = get_queries_fn(*args, **kwargs)
+        self.cache.put(cache_path, result)
+        return result
+
+
+class EmptySystemMetricsSource:
+    """Empty system metrics source that can be used as a default. Just returns an empty list of system metrics
+    for any resource."""
+
+    def __init__(self, *args, **kwargs):
+        kwargs.pop("session", None)
+        if len(args) > 0:
+            logger.warning("Received unexpected arguments: %s", args)
+        if len(kwargs) > 0:
+            logger.warning("Received unexpected keyword arguments: %s", kwargs)
+        super().__init__()
```

Comment:
Any reason to not define a strong interface here directly __init__(self, session)? 

Also, I think `super().__init__()` will be redundant here.

Reply:
if this is only to keep a generic enough default, i'd remove the init 🤷

Reply:
in any case noone would inherit this no? if we jump into non-SQA sources then they'd need their own class and whatnot